### PR TITLE
Build url if repository url field is null

### PR DESCRIPTION
**Pull request by kjlundsgaard**
Creates a url to link to if the repository url returns null

cc @benvinegar

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releases/releaseCommits.jsx [hunk 1]
```diff
@@ -29,7 +32,10 @@ const ReleaseCommit = React.createClass({
           <div className="col-xs-2"><span className="repo-label">{this.props.repository.name}</span></div>
           <div className="col-xs-2 align-right">
             <a className="btn btn-default btn-sm"
-               href={this.props.repository.url + '/' + this.props.commitId}
+               href={this.props.repository.url
+                ? (this.props.repository.url + '/' + this.props.commitId)
+                : ('http://' + this.props.repository.provider.id + '.com/' + repoPath + '/' + this.props.commitId)
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/releases/releaseCommits.jsx]: I just realized it's `/commit/SHA`, so we'll need to change this.

File: src/sentry/static/sentry/app/views/releases/releaseCommits.jsx [hunk 2]
```diff
@@ -18,6 +18,9 @@ const ReleaseCommit = React.createClass({
 
   },
   render() {
+    let name = this.props.repository.name;
+    let match = name.match(/:(.+)/);
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/releases/releaseCommits.jsx]: It's probably safer if this match failed gracefully if the `provider:` prefix was absent. (If there is no provider, this `match` call will return `null` and `match[1]` will throw an exception.

I'd do:

```javascript
let match = name.match(/[^:]+$/);
let repoPath = match[0];
```

Results:

```javascript
'github:getsentry/sentry'.match(/[^:]+$/);
// getsentry/sentry
'getsentry/sentry'.match(/[^:]+$/);
// getsentry/sentry
':getsentry/sentry'.match(/[^:]+$/);
// getsentry/sentry
```

File: src/sentry/static/sentry/app/views/releases/releaseCommits.jsx [hunk 3]
```diff
@@ -29,7 +32,10 @@ const ReleaseCommit = React.createClass({
           <div className="col-xs-2"><span className="repo-label">{this.props.repository.name}</span></div>
           <div className="col-xs-2 align-right">
             <a className="btn btn-default btn-sm"
-               href={this.props.repository.url + '/' + this.props.commitId}
+               href={this.props.repository.url
+                ? (this.props.repository.url + '/' + this.props.commitId)
+                : ('http://' + this.props.repository.provider.id + '.com/' + repoPath + '/' + this.props.commitId)
+               }
                target="_blank"><span
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/releases/releaseCommits.jsx]: I think we should move all of this URL logic to a helper method, e.g. `getCommitURL`.

And I'd special case everything. We have no idea if Bitbucket/Stash follows the same pattern.

Something like:

```javascript
getCommitURL() {
  switch (provider) {
    case 'github':
      return url ? url + '/commit/' + sha :
        'https://github.com/' + repoPath + '/commit/' + sha;
    default:
      return url;
  }
}
```

^Something like that (totally busted).

**Review by benvinegar — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/views/releases/releaseCommits.jsx (hunk 1 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/views/releases/releaseCommits.jsx]: can you explain why we're doing this? We should fix the URL in the db.

File: src/sentry/static/sentry/app/views/releases/releaseCommits.jsx (hunk 2 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/views/releases/releaseCommits.jsx]: name is not guaranteed to be the repo name

**Pull request closed without merging**